addons/survey/models/survey_survey.py

```diff
@@ -508,6 +564,27 @@ def _get_survey_questions(self, answer=None, page_id=None, question_id=None):
             questions = questions & answer.predefined_question_ids
         return questions, page_or_question_id
 
+    # ------------------------------------------------------------
+    # CONDITIONAL MANAGEMENT
+    # ------------------------------------------------------------
+    def _get_conditional_maps(self):
```

Comment:
I would make this method use and return recordsets. Not necessarily for keys, but at least for content. That way it is easier to use in classic code. If send to frontend, simply rebuild a new mapping using records.ids . That way backend code is always using record sets (for example in _get_inactive_conditional_questions ).